### PR TITLE
Migrate tf_backend

### PR DESCRIPTION
**Pull request by psybuzz**
Checked that `bazel build tensorboard/components/tf_backend` successfully builds.

### CODE REVIEW

**Comment by psybuzz** [comment]
I've removed .prettierignore and package-lock.json.

File: tensorboard/components/tf_backend/tf-backend.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+export * from "../tf-imports/polymer.html";
+export * from "../tf-imports/lodash.html";
+export * from "../vz-sorting/vz-sorting.html";
+export * from "requestManager";
+export * from "urlPathHelpers";
+export * from "router";
```
Comment (stephanwlee) [inline tensorboard/components/tf_backend/tf-backend.ts]: please fix this file, too.

Comment (psybuzz) [reply tensorboard/components/tf_backend/tf-backend.ts]: Sorry, I forgot about that.  I've renamed it to `index.ts`, removed the non-backend exports, and made the module names relative paths.

File: tensorboard/components/tf_backend/index.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the 'License');
```
Comment (stephanwlee) [inline tensorboard/components/tf_backend/index.ts]: In the license, the quotes should be double quotes. There are few rare instances when that is not true and you probably copied from the wrong file :(

Comment (psybuzz) [reply tensorboard/components/tf_backend/index.ts]: Done

**Pull request merged**